### PR TITLE
Cabal new-build based Travis CI job

### PR DESCRIPTION
**Pull request by hvr**

### CODE REVIEW

**Comment by 3noch** [comment]
@hvr I didn't spend very long tuning the version bounds so I wouldn't be surprised if we end up hitting something.

File: ziptastic-client/ziptastic-client.cabal [hunk 1]
```diff
@@ -27,7 +27,7 @@ library
   hs-source-dirs:  src
   exposed-modules: Ziptastic.Client
   build-depends:
-      base                  >= 4.7 && < 5
+      base                  >= 4.8 && < 5
```
Comment (hvr) [inline ziptastic-client/ziptastic-client.cabal:30]: I just saw #8, so one might want to undo this....

**Comment by hvr** [comment]
@3noch I'm just waiting for this to get green, and then you should be good to go and pick up the `.travis.yml` file (this is the simple/basic version -- I'd file new PRs to fine-tune it, but since you seem to be eager to have CI working I suggest you start with this simpler one).

Note that after merging this PR, the travis-cache for `master` will be empty; the first build w/o a populated cache will take _significantly_ longer; once there is cached data, the job will usually only take a few minutes.

**Comment by 3noch** [comment]
@hvr I'm not in a big hurry really. Would you prefer to keep working on this?

BTW: Thank you!

**Comment by 3noch** [comment]
🎉

**Comment by hvr** [comment]
@3noch well, I need to take care of something else for the rest of the day, so it's best if you take the current state... :-)

**Comment by 3noch** [comment]
@hvr Sounds good. This is so helpful! I hope you learned what you wanted to learn in doing this.

**Pull request merged**